enacts/calc.py

```diff
@@ -345,39 +382,63 @@ def cess_date(
 
     Parameters
     ----------
-    soil_moisture : DataArray
-        Array of daily soil moisture.
+    daily_data : DataArray
+        Array of daily soil moisture or daily rainfall.
     dry_thresh : float
-        Soil moisture threshold to determine dry day if `dry_thresh` is less than `soil_moisture`.
-    min_dry_days : int
+        Soil moisture threshold to determine a dry day.
+    dry_spell_length_thresh : int
         Minimum number of dry days in a row to be considered a dry spell.
-    time_coord : str, optional
-        Time coordinate in `soil_moisture` (default `time_coord`="T"). 
+    et : DataArray, optional
+        Evapotranspiration used by `water_balance` if `daily_data` is rainfall.
+        Can be a single value with no dimensions or axes.
+    taw : DataArray, optional
+        Total available water used by `water_balance` if `daily_data` is rainfall.
+        Can be a single value with no dimensions or axes.
+    sminit : DataArray, optional
+        Soil moisture initialization used by `water_balance` if `daily_data` is rainfall.
+        If DataArray, must not have `time_dim` dim.
+        Can be a single value with no dimensions or axes.
+    time_dim : str, optional
+        Time coordinate in `soil_moisture` (default `time_dim`="T"). 
+    
     Returns
     -------
     cess_delta : DataArray[np.timedelta64] 
-        Difference between first day of `soil_moisture` 
+        Difference between first day of `daily_data` 
         and cessation date.
+    
     See Also
     --------
+    water_balance
+
     Notes
     -----
     Examples
     --------
     """
-    dry_day = soil_moisture < dry_thresh
-    dry_spell = dry_day * 1
-    dry_spell_roll = dry_spell.rolling(**{time_coord: min_dry_days}).sum() == min_dry_days
-    cess_mask = dry_spell_roll * 1
-    cess_mask = cess_mask.where((cess_mask == 1))
-    cess_delta = cess_mask.idxmax(dim=time_coord)
+    # Initializing
+    spell_length = (
+        daily_data.isel({time_dim: 0}).expand_dims(dim=time_dim) < dry_thresh
+    ).astype("timedelta64[D]")
+    cess_delta = xr.DataArray(np.timedelta64("NaT", "D"))
+    # Loop
+    for t in daily_data[time_dim][1:]:
+        dry_day = daily_data.sel({time_dim: t}).expand_dims(dim=time_dim) < dry_thresh
+        spell_length = (
+            spell_length.squeeze(time_dim, drop=True) + dry_day.astype("timedelta64[D]")
+        ) * dry_day
+        cess_delta = cess_date_step(
+            cess_delta.squeeze(drop=True),
+            spell_length,
+            np.timedelta64(dry_spell_length_thresh, "D"),
+        )
+    # Delta reference (and coordinate) back to first time point of daily_data
```

Comment:
I don't follow why all the squeezing and expanding is necessary, but I'll take your word for it. If it's just leftover from some previous version, try removing it.

Reply:
It was not a leftover but can indeed do without and add the T we want at the end of the function only. It might not fly when the water balance option kicks in but might as well start from something simpler